docs/book/driver-deployment/upgrade.md

```diff
@@ -0,0 +1,30 @@
+<!-- markdownlint-disable MD033 -->
+# vSphere CSI Driver - Upgrade
+
+**Note: To upgrade "vSphere with Tanzu" follow vSphere documentation. Following instructions are applicable to native vanilla Kubernetes Cluster deployed on vSphere.**
+
+Each release of vSphere CSI driver has a different set of RBAC rules and deployment YAML files depending on the version of the vCenter on which Kubernetes Cluster is deployed.
+
+To upgrade from one release to another release of vSphere CSI driver, we recommend to completely delete the driver using the original YAML files, you have used to deploy the driver and then use new YAML files to re-install the vSphere CSI driver.
```

Comment:
Why do we recommend something like this -  delete and create new one?
CSI driver deployed on Production would cause a possible downtime for some running applications.

Since we use standard K8S constructs, can we make use of K8S update strategies like RollingUpgrade etc.

Reply:
rolling upgrade is good for application upgrade where zero downtime is required and it is ok to have some pods running with a prior release.

Delete and Recreate is the clean approach to make sure all previous resources are deleted and new resources are created from newer YAML files.

Also, we do not want customers to use rolling upgrade to use some pods with prior release of driver and some pods with a newer release of the driver. we want all pods to be re-created with newer images.

Reply:
>Also, we do not want customers to use rolling upgrade to use some pods with prior release of driver and some pods with a newer release of the driver. 

but we only have one replica which means one pod.

Reply:
for multi-master setup, I have seen customers are setting this replica count to the number of masters they have in the system, and also we have support for `leader-election`, so we can increase replica counts.

Reply:
okay.